packages/react-native-codegen/src/parsers/typescript/components/events.js

```diff
@@ -216,49 +197,38 @@ function getEventArgument(argumentProps, name: $FlowFixMe) {
   };
 }
 
-function findEvent(typeAnnotation: $FlowFixMe, optional: boolean) {
+function isEvent(typeAnnotation: $FlowFixMe) {
   switch (typeAnnotation.type) {
-    // Check for T | null | undefined
-    case 'TSUnionType':
-      return findEvent(
-        typeAnnotation.types.filter(
-          t => t.type !== 'TSNullKeyword' && t.type !== 'TSUndefinedKeyword',
-        )[0],
-        optional ||
-          typeAnnotation.types.some(
-            t => t.type === 'TSNullKeyword' || t.type === 'TSUndefinedKeyword',
-          ),
-      );
-    // Check for (T)
-    case 'TSParenthesizedType':
-      return findEvent(typeAnnotation.typeAnnotation, optional);
     case 'TSTypeReference':
       if (
         typeAnnotation.typeName.name !== 'BubblingEventHandler' &&
         typeAnnotation.typeName.name !== 'DirectEventHandler'
       ) {
-        return null;
+        return false;
       } else {
-        return {typeAnnotation, optional};
+        return true;
       }
     default:
-      return null;
+      return false;
```

Comment:
Can't we refactor this in simply:
```js
const eventTypes = new Set(['BubblingEventHandler' , 'DirectEventHandler']);
return typeAnnotation.type == 'TSTypeReference' && eventTypes.has(typeAnnotation.type)
```
WDYT?

Reply:
There is another similar test in `componentsUtils.js`, I will do a refactor and merge them into a single test later, code will get deleted, so I don't worry about here.